File: frontend/controle/src/app/services/sistema.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Sistema } from "../domain/entities/sistema"
import { API_SISTEMAS } from "../config/api"
import { environment } from "../../environments/environment";

@Injectable({
  providedIn: 'root'
})
export class SistemaService {

  constructor(
      private http: HttpClient
  ) { }

  public listar(): Observable<Sistema[]> {
    return this.http.get<Sistema[]>(`${environment.urlBackend}${API_SISTEMAS}`);
  }

  public buscar(id: string): Observable<Sistema> {
    return this.http.get<Sistema>(`${environment.urlBackend}${API_SISTEMAS}/${id}`);
  }

  public gravar(sistema: any): Observable<Sistema> {
    return this.http.post<Sistema>(`${environment.urlBackend}${API_SISTEMAS}`, sistema);
  }

  public excluir(id: number): Observable<any> {
    return this.http.delete(`${environment.urlBackend}${API_SISTEMAS}/${id}`);
  }

}

File: backend/core/src/main/java/com/github/andrepenteado/sso/core/services/SistemaService.java
package com.github.andrepenteado.sso.core.services;

import com.github.andrepenteado.sso.core.domain.entities.Sistema;
import org.springframework.validation.BindingResult;

import java.util.List;
import java.util.Optional;

public interface SistemaService {

    List<Sistema> listar();

    Optional<Sistema> buscar(Long id);

    Sistema incluirOuAlterar(Sistema sistema, BindingResult validacao);

    void excluir(Long id);

}
